TACC/Core-Styles
# FP-1648: Build Individual Stylesheets — ⚠️ API Change

## Overview

<details><summary>Allow consuming and/or building* individual stylesheets.</summary>

- _(for clients that only consume)_
    Build all stylesheets and commit `dist/` directory ([for Portal](https://github.com/TACC/Core-Portal/pull/639)).
- _(for clients building from source)_
    Let clients (and [this repo itself now](https://github.com/TACC/Core-Styles/pull/23/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R17)) build individual files (theirs, ours, both).*

<sub>\* CMS is the only client app that does, and is expected to, [build its styles via the Core-Styles build command](https://github.com/TACC/Core-CMS/blob/task/fp-1648-core-styles/bin/build-css-via-cli.js#L54-L59).</sub>

</details>

## Related

- [FP-1648](https://jira.tacc.utexas.edu/browse/FP-1648)
- required by https://github.com/TACC/Core-Portal/pull/639

## Changes

Top of diff and bottom of diff have the most relevant changes. In between is all `dist/`.

- e00d6b8 let user match pattern, drop `--file-ext`
- ef2db2b build all stylesheets and commit dist/
- 12f1484 use `--input`/`--ouput` not `--input-dir`/`--output-dir`
- c543a07 allow build with custom glob input
- update `.gitignore`

## Testing

1. Build all styles internally via `npm run build`.
    - Verify all are added in the same sub-dirs as `source/_imports/`.
    - Verify `dist/tools/` stylesheets have either class selectors (`.bob`) or functional selectors (`%bob`), not both.
3. Test all styles internally via `npm run test` (builds output of plugin tests).
    - Verify build output in `dist/_tests/`.
4. Test https://github.com/TACC/Core-CMS/pull/489/files.

## Screenshots

N/A
